project/JavalibIRCleaner.scala

```diff
@@ -107,14 +114,42 @@ final class JavalibIRCleaner(baseDirectoryURI: URI) {
     def errorCount: Int = _errorCount
   }
 
-  private def readIR(file: File): ClassDef = {
-    import java.nio.ByteBuffer
+  private def readIR(file: File): ClassDef =
+    readIR(file.toPath())
 
-    val bytes = Files.readAllBytes(file.toPath())
+  private def readIR(path: Path): ClassDef = {
+    val bytes = Files.readAllBytes(path)
     val buffer = ByteBuffer.wrap(bytes)
     Serializers.deserialize(buffer)
   }
 
+  private def readIRJar(jar: File): List[ClassDef] = {
+    // Similar to PathIRContainer.JarIRContainer and its walkIR helper
+
+    val classDefs = List.newBuilder[ClassDef]
+
+    val dirVisitor = new SimpleFileVisitor[Path] {
+      override def visitFile(path: Path, attrs: BasicFileAttributes): FileVisitResult = {
+        if (path.getFileName().toString().endsWith(".sjsir"))
+          classDefs += readIR(path)
+        super.visitFile(path, attrs)
+      }
+    }
+
+    // Open zip/jar file as filesystem.
+    // The type ascription is necessary on JDK 13+.
+    val fs = FileSystems.newFileSystem(jar.toPath(), null: ClassLoader)
+    try {
+      val iter = fs.getRootDirectories().iterator()
+      while (iter.hasNext())
+        Files.walkFileTree(iter.next(), dirVisitor)
```

Comment:
I just realized: We should probably use `Files.walk` here, because we do not need `attrs`.

I'm happy to follow-up on this; I might take the opportunity to rewrite the entire thing using streams. Seems like it might actually give us some memory benefits.